third_party/blink/renderer/bindings/core/v8/custom/v8_window_custom.cc

```diff
@@ -251,7 +251,15 @@ void V8Window::NamedPropertyGetterCustom(
   }
 
   auto* base_doc = To<LocalFrame>(frame)->GetDocument();
-  recordreplay::Diagnostic("[RUN-1747-1792] V8Window::NamedPropertyGetterCustom #5 %p", base_doc);
+
+  // LocalFrame document pointers can be invalid and lead to crashes while replaying
+  // and inspecting the current state. The underlying reason here is unknown,
+  // for now we workaround the problem by watching for invalid document pointers.
+  // Documents register themselves so we can test for a pointer ID.
+  if (recordreplay::IsInReplayCode() && base_doc && !recordreplay::PointerId(base_doc)) {
```

Comment:
Playing devil's advocate: If the pointer is corrupt, what is the chance of this coincidentally pointing to something else that we actually have registered or it being zeroed out?

Reply:
There is some chance, but it is much less.  The problem with corruption is that you can never be 100% sure the mitigations you've put into place are correct -- even if we knew this was a valid document pointer we couldn't be sure it's the *right* document pointer.  The actual solution is to figure out the reason for the corruption, but crashing less often is a nice intermediate goal.